projects/components/src/table/table.component.ts

```diff
@@ -562,6 +571,10 @@ export class TableComponent
         [TableComponent.PAGE_SIZE_URL_PARAM]: pageEvent.pageSize
       });
     }
+
+    this.selections = [];
+    this.selectionsChange.emit(this.selections);
```

Comment:
This should only emit if there were previously selections - that is, it changed.

Reply:
Just for optimization, right?

Reply:
For correctness - this is a change emitter, it should only emit on change.